### PR TITLE
Add Key vault as spring property source

### PR DESCRIPTION
**Pull request by warrenzhu25**

### CODE REVIEW

File: spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 @Configuration
 @EnableConfigurationProperties(AzureProperties.class)
 @ConditionalOnClass(Azure.class)
-@ConditionalOnProperty(value = "spring.cloud.azure.enabled", matchIfMissing = true)
+@ConditionalOnProperty(value = "spring.cloud.azure.credentialFilePath")
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java]: we will refine property name to some thing like `credential-file-path`, right ?

File: spring-cloud-azure-keyvault/src/main/java/com/microsoft/azure/spring/cloud/keyvault/KeyVaultEnvironmentPostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    private static boolean keyVaultPropertiesProvided(ConfigurableEnvironment environment) {
+        return AZURE_KEY_VAULT_PROPERTIES.stream().allMatch(environment::containsProperty);
+    }
+
+    private static boolean isKeyVaultClientAvailable() {
+        return ClassUtils.isPresent("com.microsoft.azure.keyvault.KeyVaultClient",
+                KeyVaultEnvironmentPostProcessor.class.getClassLoader());
+    }
+
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-keyvault/src/main/java/com/microsoft/azure/spring/cloud/keyvault/KeyVaultEnvironmentPostProcessor.java]: too many empty line here.

File: spring-cloud-azure-keyvault/src/main/java/com/microsoft/azure/spring/cloud/keyvault/KeyVaultPropertySource.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        super("KeyVault", operation);
+        this.operation = operation;
+        this.keyVaultName = keyVaultName;
+    }
+
+    public String[] getPropertyNames() {
+        return this.operation.listSecrets(this.keyVaultName).toArray(new String[0]);
+    }
+
+    public Object getProperty(String name) {
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-keyvault/src/main/java/com/microsoft/azure/spring/cloud/keyvault/KeyVaultPropertySource.java]: As public method, suggest to add `@Nullable` to guide user.

File: spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java (hunk 1 above)
Comment (warrenzhu25) [reply spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java]: Yes, in another PR.

File: spring-cloud-azure-keyvault/src/main/java/com/microsoft/azure/spring/cloud/keyvault/KeyVaultEnvironmentPostProcessor.java (hunk 2 above)
Comment (warrenzhu25) [reply spring-cloud-azure-keyvault/src/main/java/com/microsoft/azure/spring/cloud/keyvault/KeyVaultEnvironmentPostProcessor.java]: Yes. Will remove it.

**Pull request merged**